secretsauce/apps/portal/models.py

```diff
@@ -3,25 +3,74 @@
 # Create your models here.
 
 class Project(models.Model):
-    title = models.CharField(max_length=200)
-    organization = models.CharField(max_length=200)
+    title = models.CharField("Project Title", max_length=200)
+    organization = models.CharField("Organization", max_length=200)
+    # TODO: Change organization into choices
+
+    def __str__(self):
+        return "%s: %s" % (self.organization, self.title)
 
 class PredictionModel(models.Model):
-    model_type = models.CharField(max_length=200)
     model_name = models.CharField(max_length=200)
+    description = models.TextField()
+
+    def __str__(self):
+        return self.model_name
+
+class ModelTag(models.Model):
+    tag_name = models.CharField(max_length=200)
+    prediction_model = models.ManyToManyField(
+        PredictionModel,
+        related_name="tags"
+    )
+    # TODO: No repeated tags
+
+    def __str__(self):
+        return self.tag_name
 
 class Hyperparameters(models.Model):
+    """
+    Required hyperparameters to train a defined PredictionModel
+    """
     prediction_model = models.ForeignKey(PredictionModel, on_delete=models.CASCADE)
+    hyperparameter_name = models.CharField(max_length=200)
+    HYPERPARAM_TYPES = (
+        ('INT', 'Integer'),
+        ('FP', 'Floating point'),
+        ('STR', 'String'),
+        ('CAT', 'Category')
+    )
+    hyperparameter_type = models.CharField(max_length=3, choices=HYPERPARAM_TYPES)
 
-class Weights(models.Model):
-    model_type = models.ForeignKey(PredictionModel, on_delete=models.CASCADE)
-    url = models.URLField(max_length = 200) 
+    def __str__(self):
+        return "%s: %s (%s)" %(self.prediction_model, self.hyperparameter_name, self.hyperparameter_type)
 
 class DataBlock(models.Model):
     project = models.ForeignKey(Project, on_delete=models.CASCADE)
+    data_block_name = models.CharField(max_length=200)
+    upload = models.FileField(upload_to="uploads/")
+
+    def __str__(self):
+        return "DataBlock: %s" % (self.data_block_name) 
+
+class Weights(models.Model):
+    prediction_model = models.ForeignKey(PredictionModel, on_delete=models.CASCADE)
+    data_block = models.ForeignKey(DataBlock, on_delete=models.CASCADE)
+    upload = models.URLField(max_length = 200) 
+
+    def __str__(self):
+        return "It's a Weight!"
```

Comment:
```suggestion
        return str(self.id)
```
If we cannot think of a unique name for each weight for now, would it be better to just return the id of the model instance instead?